src/resolve/resolved-dns-scope.c

```diff
@@ -306,10 +306,11 @@ static int dns_scope_socket(
                 int family,
                 const union in_addr_union *address,
                 DnsServer *server,
-                uint16_t port) {
+                uint16_t port,
+                union sockaddr_union *ret_socket_address) {
```

Comment:
a quick comment somewhere which explains this "delayed connect logic" a bit, i.e. that the caller needs to do the connect on its own later on (and why he would choose to do so) would be nice.

Reply:
a comment is added to the non-static method dns_scope_socket_tcp